encoding/ssz/htrutils_test.go

```diff
@@ -122,6 +122,43 @@ func TestTransactionsRoot(t *testing.T) {
 	}
 }
 
+func TestByteSliceRoot(t *testing.T) {
+	tests := []struct {
+		name    string
+		slice   []byte
+		want    [32]byte
+		wantErr bool
+	}{
+		{
+			name:  "nil",
+			slice: nil,
+			want:  [32]byte{148, 207, 155, 226, 2, 65, 69, 197, 173, 124, 141, 137, 63, 194, 41, 46, 78, 190, 32, 126, 164, 35, 80, 252, 124, 243, 232, 121, 138, 195, 76, 217},
```

Comment:
Should nil not be an error?

Reply:
That's why i wanted to double check with you and potuz this was the case for TestTransactionsRoot 

I didn't change any of the business logic it does not error

Reply:
It looks like @terencechain added this func, I'm not sure if we should treat nil values as zero value or not.

Reply:
Empty lists are allowed, not sure if it's reasonable to separate nil from empty